src/main/scala/com/raphtory/spouts/HdfsSpout.scala

```diff
@@ -0,0 +1,55 @@
+package com.raphtory.spouts
+
+import com.raphtory.core.components.spout.Spout
+
+import java.io.{BufferedReader, InputStreamReader}
+import java.net.URI
+
+import org.apache.hadoop.conf.Configuration
+import org.apache.hadoop.fs.{FileSystem, Path}
+
+class HdfsSpout(headerUrlPort:String, route:String, dropHeader:Boolean=false) extends Spout[String] {
+
+  val fs = FileSystem.get(new URI(headerUrlPort), new Configuration())
+  private var status = fs.listStatus(new Path(route))
+  private var hdfsManager = HDFSManager()
+
+  override def setupDataSource(): Unit = {}
+
+  override def closeDataSource(): Unit = {}
+
+  override def generateData(): Option[String] = {
+    val line = hdfsManager.nextLine()
+    if (hdfsManager.finished(line)) {
+      dataSourceComplete()
+      None
+    } else {
+      Some(line)
+    }
+  }
+
+  final case class HDFSManager private () {
+    val stream = fs.open(status(0).getPath)
+    status = status.slice(1,status.length)
+    val bufferReader = new BufferedReader(new InputStreamReader(stream))
+
+    def nextLine(): String = {
+      bufferReader.readLine()
+    }
+
+    def finished(line: String): Boolean = {
+      if (line == null) {
+        if(status.length > 0) {
+          hdfsManager = HDFSManager()
+          println("Still loading Spout...")
```

Comment:
Hi Rodrigo, thanks for adding this in! Would you mind changing these prints to use loggers and I shall merge it in :)

Reply:
Done! I hope it is what you were hoping for.

Reply:
Fab! Will merge it in now 🎉

Reply:
Nice! :)